src/s390x/zipl.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::dasd::check;
+use crate::errors::*;
+use regex::Regex;
+use std::fs::read_to_string;
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tempfile::NamedTempFile;
+use walkdir::WalkDir;
+
+/////////////////////////////////////////////////////////////////////////////
+// IBM Z Bootloader Support
+/////////////////////////////////////////////////////////////////////////////
+
+/// Generate zipl config and set dasd as boot device
+///
+/// # Arguments
+/// * `boot` - path to boot partition mountpoint, i.e. smth like /boot
+/// * `dasd` - dasd device, i.e. smth like /dev/dasda
+pub fn install_bootloader<P: AsRef<Path>>(
+    boot: P,
+    dasd: &str,
+    firstboot_kargs: &Option<String>,
+) -> Result<()> {
+    eprintln!("Installing bootloader");
+
+    let bootconf = boot_config(&boot)?;
+    let vmlinuz = kernel(&boot)?;
+    let ramfs = initramfs(&boot)?;
+
+    update_boot_config(&bootconf)?;
+
+    let mut zipl_config = NamedTempFile::new().chain_err(|| "creating zipl config file")?;
+    generate_zipl_config(&bootconf, &mut zipl_config, &firstboot_kargs)
+        .chain_err(|| "writing boot options")?;
+
+    let status = Command::new("zipl")
+        .arg("--verbose")
+        .arg("-p")
+        .arg(zipl_config.path())
+        .arg("-i")
+        .arg(vmlinuz.to_str().unwrap())
```

Comment:
This is probably another case where we want to use
`vmlinuz.as_os_str()`

Reply:
done